src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -1549,7 +1663,44 @@ splitSuffixBy predicate f = foldMany1 (FL.sliceSepBy predicate f)
 
 {-# INLINE_NORMAL wordsBy #-}
 wordsBy :: Monad m => (a -> Bool) -> Fold m a b -> Stream m a -> Stream m b
-wordsBy predicate f = foldMany1 (FL.wordBy predicate f)
+wordsBy predicate f (Stream step state) = Stream (stepOuter f) (Just state)
```

Comment:
We can define it using parseMany and compare if perf remains the same.

Reply:
We will do it in a future change. The parser functions are still not defined I believe.